packages/hash/subgraph/src/stdlib/edge/link.ts

```diff
@@ -63,6 +65,88 @@ export const getOutgoingLinksForEntityAtMoment = (
   );
 };
 
+/**
+ * For a given moment in time, get all incoming link entities from a given entity.
+ *
+ * @param subgraph
+ * @param {EntityId} entityId - The ID of the source entity to search for outgoing links from
+ * @param {Date | string} timestamp - A `Date` or an ISO-formatted datetime string of the moment to search for
+ * @param includeArchived - Whether or not to return currently-archived links, defaults to false
+ */
+export const getIncomingLinksForEntityAtMoment = (
+  subgraph: Subgraph,
+  entityId: EntityId,
+  timestamp: Date | string,
+  includeArchived: boolean = false,
+): EntityWithMetadata[] => {
+  const timestampString =
+    typeof timestamp === "string" ? timestamp : timestamp.toISOString();
+
+  /**
+   * @todo - replace this with the analogue of the code in `getOutgoingLinksForEntityAtMoment` when
+   *   reverse mappings are stored in the graph
+   *   https://app.asana.com/0/1201095311341924/1203399511264512/f
+   */
+  return Object.entries(subgraph.edges)
+    .filter(([_, inner]) => {
+      return Object.values(inner)
+        .flat()
+        .find(
+          (outwardEdge: OutwardEdge) =>
+            isHasRightEntityEdge(outwardEdge) &&
+            outwardEdge.rightEndpoint.baseId === entityId,
+        );
+    })
+    .map(([linkEntityId, _]) => {
+      const linkEntity = mustBeDefined(
+        getEntityAtTimestamp(
+          subgraph,
+          linkEntityId as EntityId,
+          // Find the edition of the link at the given moment (not at `_firstEditionTimestamp`, the start of its history)
```

Comment:
Comment here was copied from the other traversal function, do we keep it and rename `_` to `_firstEditionTimestamp` above?

Reply:
Correct, hm. `_` here is actually the object `{ _firstEditionTimestamp, outwardEdges[]}`, and I don't think it'd be worth destructuring the object just to have an unused variable in the comment.

I also don't really love the idea of deleting the comment. I'll try and rework it

Reply:
I've reworked the comment now :)